File: examples/governance_static_go/domain/suspended_policy.go
package domain

import "time"

type SuspendedPolicy struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	PolicyId string `json:"policy_id"`
	Name string `json:"name"`
	Description string `json:"description"`
	Category string `json:"category"`
	FrameworkId string `json:"framework_id"`
	EffectiveDate time.Time `json:"effective_date"`
	ReviewDate time.Time `json:"review_date"`
	Requirements []Requirement `json:"requirements"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e SuspendedPolicy) EventName() string { return "SuspendedPolicy" }

func (e SuspendedPolicy) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/compliance_domain/lib/compliance_domain/governance_policy/events/suspended_policy.rb
module ComplianceDomain
  class GovernancePolicy
    module Events
      class SuspendedPolicy
        attr_reader :aggregate_id, :policy_id, :name, :description, :category, :framework_id, :effective_date, :review_date, :requirements, :status, :occurred_at

        def initialize(aggregate_id: nil, policy_id: nil, name: nil, description: nil, category: nil, framework_id: nil, effective_date: nil, review_date: nil, requirements: nil, status: nil)
          @aggregate_id = aggregate_id
          @policy_id = policy_id
          @name = name
          @description = description
          @category = category
          @framework_id = framework_id
          @effective_date = effective_date
          @review_date = review_date
          @requirements = requirements
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
